helix-view/src/document.rs

```diff
@@ -774,8 +774,14 @@ impl Document {
         {
             use std::process::Stdio;
             let text = self.text().clone();
+            let cwd = std::env::current_dir().unwrap_or(PathBuf::from("."));
+            let doc_path = self.path.clone().unwrap_or_else(|| cwd.join(self.display_name().to_string()));
+            let doc_dir = doc_path.parent().unwrap_or(cwd.as_path());
+
             let mut process = tokio::process::Command::new(&fmt_cmd);
```

Comment:
We can avoid setting `current_dir` when the document doesn't have a path (i.e. is a scratch buffer) so that the process is started from the CWD instead.

```rust
if let Some(doc_dir) = self.path.as_ref().and_then(|path| path.parent()) {
    process.current_dir(doc_dir);
}
```

Reply:
Ahh nice thank you! Tested locally and this works well too. Updated PR

I thought since `process.current_dir` returns `&mut Command` that maybe it needed to be chained.

Reply:
Yeah it's a cool idiom that you can write an interface as `fn foo(&mut self) -> &mut Self` and use it to chain calls or to make an update in place